### PR TITLE
Scrs 2916 officers home addres page

### PR DESCRIPTION
**Pull request by darrell-plant**

### CODE REVIEW

File: app/connectors/PPConnector.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ import scala.concurrent.ExecutionContext.Implicits.global
 @ImplementedBy(classOf[PrePopConnector])
 trait PPConnector {
 
-  val companyRegUrl: String
+//  val companyRegUrl: String
```
Comment (PeterPerhac) [inline app/connectors/PPConnector.scala]: i assume there is still an intention to un-commented these lines. 👍

File: app/controllers/VatRegistrationController.scala [hunk 2]
```diff
@@ -22,6 +22,7 @@ import auth.VatTaxRegime
 import cats.data.OptionT
 import cats.instances.future._
 import config.FrontendAuthConnector
+import connectors.KeystoreConnector
```
Comment (PeterPerhac) [inline app/controllers/VatRegistrationController.scala]: here seems to be just an import added without any code change requiring the import

File: app/controllers/vatLodgingOfficer/OfficerHomeAddressController.scala [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  import cats.instances.future._
+
+  val form = OfficerHomeAddressForm.form
+
+  def show: Action[AnyContent] = authorised.async(implicit user => implicit request => {
+    for {
+      addresses <- prePopService.getOfficerAddressList()
+      _ <- keystoreConnector.cache[Seq[ScrsAddress]]("OfficerAddressList", addresses)
+      res <- viewModel[OfficerHomeAddressView].fold(form)(form.fill)
+        .map(f => Ok(views.html.pages.vatLodgingOfficer.officer_home_address(f, addresses)))
```
Comment (PeterPerhac) [inline app/controllers/vatLodgingOfficer/OfficerHomeAddressController.scala]: .map inside the for{ ... } block and yield just returns res. This should be (i think)

```
vm <- viewModel....
} yield Ok(views.html.pages.vatLodgingOfficer.officer_home_address(vm, addresses))
```

File: app/models/S4LModels.scala [hunk 4]
```diff
@@ -83,3 +84,9 @@ case class S4LVatEligibility
 (
   vatEligibility: Option[VatServiceEligibility]
 )
+
+case class S4LVatLodgingOfficer
+(
+  officerHomeAddressView: Option[OfficerHomeAddressView]
```
Comment (PeterPerhac) [inline app/models/S4LModels.scala]: i assume we are adding more elements to the case class soon. we don't want single-element product types

File: app/models/external/CoHoRegisteredOfficeAddress.scala [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                        postalCode: Option[String],
+                        region: Option[String]
+                      )
+
+object CoHoRegisteredOfficeAddress {
+
+  private val roAddress = __ \ "registered_office_address"
+
+  implicit val formatModel: Reads[CoHoRegisteredOfficeAddress] = (
+      (roAddress \ "premises").read[String] and
```
Comment (PeterPerhac) [inline app/models/external/CoHoRegisteredOfficeAddress.scala]: 👍

File: app/controllers/VatRegistrationController.scala (hunk 2 above)
Comment (darrell-plant) [inline app/controllers/VatRegistrationController.scala]: Ah, I added a Keystore a few commits ago, then removed but didn't fix imports. Fixing...

File: app/connectors/PPConnector.scala (hunk 1 above)
Comment (darrell-plant) [inline app/connectors/PPConnector.scala]: I had to comment or remove these lines, as I wanted to make use of:
  val companyRegUrl = baseUrl("company-registration")
in CompanyRegistrationConnector

will explain tomorrow

File: app/controllers/vatLodgingOfficer/OfficerHomeAddressController.scala (hunk 3 above)
Comment (darrell-plant) [inline app/controllers/vatLodgingOfficer/OfficerHomeAddressController.scala]: Controller still needs work. Tomorrow...

File: app/models/S4LModels.scala (hunk 4 above)
Comment (darrell-plant) [inline app/models/S4LModels.scala]: We are.

**Pull request merged**